spark/tools/generate_glove_training_data.py

```diff
@@ -0,0 +1,91 @@
+import pyspark.sql.functions as F
+import pyspark.sql.types as T
+import datetime
+from pyspark.sql import SparkSession
+from pyspark.sql import Window
+
+def join_domain_tables(domain_tables):
+
+    patient_event = None
+
+    for domain_table in domain_tables:
+        #extract the domain concept_id from the table fields. E.g. condition_concept_id from condition_occurrence
+        concept_id_field = [f.name for f in domain_table.schema.fields if "concept_id" in f.name][0]
+
+        #extract the domain start_date column
+        date_field = [f.name for f in domain_table.schema.fields if "date" in f.name][0]
+
+        #extract the name of the table
+        table_domain_field = concept_id_field.replace("_concept_id", "")
+
+        domain_table = domain_table.withColumn("date", F.unix_timestamp(F.to_date(F.col(date_field)), "yyyy-MM-dd")) \
+            .select(domain_table["person_id"],
+                    domain_table[concept_id_field].alias("standard_concept_id"),
+                    F.col("date"),
+                    F.lit(table_domain_field).alias("domain"))
+
+        if patient_event == None:
+            patient_event = domain_table
+        else:
+            patient_event = patient_event.union(domain_table)
+
+    return patient_event
+
+def roll_up_to_drug_ingredients(patient_event, concept, concept_ancestor):
+
+    drug_ingredient = patient_event.where('domain="drug"').select('standard_concept_id').distinct() \
+        .join(concept_ancestor, F.col('standard_concept_id') == F.col('descendant_concept_id')) \
+        .join(concept, F.col('ancestor_concept_id') == F.col('concept_id')) \
```

Comment:
I am thinking this might be  `.join(concept, F.col('descendant_concept_id') == F.col('concept_id'))` which will give you the ingredient level concept code?